server/src/main/java/org/elasticsearch/index/IndexMode.java

```diff
@@ -249,8 +249,18 @@ public void validateTimestampFieldMapping(boolean isDataStream, MappingLookup ma
         }
 
         @Override
-        public CompressedXContent getDefaultMapping() {
-            return DEFAULT_LOGS_TIMESTAMP_MAPPING;
+        public CompressedXContent getDefaultMapping(final IndexSettings indexSettings) {
+            final IndexSortConfig.FieldSortSpec[] fieldSortSpecs = indexSettings.getIndexSortConfig().sortSpecs;
+            if (fieldSortSpecs == null || fieldSortSpecs.length == 0) {
```

Comment:
I think we don't need this check because the index sort for logsdb should already be injected in the IndexSortConfig constructor.